services/datamanager/builtin/builtin.go

```diff
@@ -677,52 +677,80 @@ func isOffline() bool {
 func (svc *builtIn) sync() {
 	svc.flushCollectors()
 
+	// Lock while retrieving any values that could be changed during reconfiguration of the data
+	// manager.
 	svc.lock.Lock()
-	toSync := getAllFilesToSync(svc.captureDir, svc.fileLastModifiedMillis)
-	for _, ap := range svc.additionalSyncPaths {
-		toSync = append(toSync, getAllFilesToSync(ap, svc.fileLastModifiedMillis)...)
-	}
-	svc.lock.Unlock()
 
+	captureDir := svc.captureDir
+	fileLastModifiedMillis := svc.fileLastModifiedMillis
+	additionalSyncPaths := svc.additionalSyncPaths
 	stopAfter := time.Now().Add(time.Duration(svc.syncIntervalMins * float64(time.Minute)))
-	for _, p := range toSync {
-		svc.syncer.SyncFile(p, stopAfter)
+
+	svc.lock.Unlock()
+
+	// Kick off a goroutine to retrieve all the names of the files to sync, then another 1000 to
+	// sync the files to Viam app.
+	numWorkers := 1000
+	fileChannel := make(chan string, numWorkers)
+
+	var wg sync.WaitGroup
```

Comment:
Open question: is it okay that I made a new waitgroup? Or should I use syncer.backgroundWorkers?

Reply:
I've resolved that I need a waitgroup here but that I should also add to backgroundWorkers for both the service and the syncer so it knows to wait for those.